src/routes/safe/components/AddressBook/CreateEditEntryModal/index.jsx

```diff
@@ -0,0 +1,126 @@
+// @flow
+import React from 'react'
+import { withStyles } from '@material-ui/core/styles'
+import Close from '@material-ui/icons/Close'
+import IconButton from '@material-ui/core/IconButton'
+import { useSelector } from 'react-redux'
+import Row from '~/components/layout/Row'
+import Block from '~/components/layout/Block'
+import GnoForm from '~/components/forms/GnoForm'
+import Button from '~/components/layout/Button'
+import Hairline from '~/components/layout/Hairline'
+import Field from '~/components/forms/Field'
+import TextField from '~/components/forms/TextField'
+import Paragraph from '~/components/layout/Paragraph'
+import {
+  composeValidators, required, minMaxLength, uniqueAddress,
+} from '~/components/forms/validator'
+import Modal from '~/components/Modal'
+import { styles } from './style'
+import AddressInput from '~/components/forms/AddressInput'
+import type { AddressBookEntryType } from '~/logic/addressBook/model/addressBook'
+import { getAddressBookAddressesListSelector } from '~/logic/addressBook/store/selectors'
+
+export const CREATE_ENTRY_INPUT_NAME_ID = 'create-entry-input-name'
+export const CREATE_ENTRY_INPUT_ADDRESS_ID = 'create-entry-input-address'
+export const SAVE_NEW_ENTRY_BTN_ID = 'save-new-entry-btn-id'
+
+type Props = {
+  onClose: () => void,
+  classes: Object,
+  isOpen: boolean,
+  newEntryModalHandler: Function,
+  editEntryModalHandler: Function,
+  entryToEdit?: AddressBookEntryType,
+}
+
+const CreateEditEntryModalComponent = ({
+  onClose,
+  isOpen,
+  classes,
+  newEntryModalHandler,
+  entryToEdit,
+  editEntryModalHandler,
+}: Props) => {
+  const onFormSubmitted = (values) => {
+    if (entryToEdit) {
+      editEntryModalHandler(values)
+    } else {
+      newEntryModalHandler(values)
+    }
+  }
+
+  const addressBookAddressesList = useSelector(getAddressBookAddressesListSelector)
+  const entryDoesntExist = uniqueAddress(addressBookAddressesList.map((address) => address))
```

Comment:
```
addressBookAddressesList.map((address) => address)
```
why do we need this? uniqueAddress uses `.some` method which both Arrays and Immutable List support

Reply:
I'd just modified that logic

Reply:
Sorry, I don't understand. Validator logic is not modified, it should be fine to just pass `addressBookAddressesList`

Reply:
Done